k8s/schemaless.go

```diff
@@ -28,6 +28,15 @@ type SchemalessClient struct {
 	mutex sync.Mutex
 }
 
+// NewSchemalessClient creates a new SchemalessClient using the provided rest.Config and ClientConfig.
+func NewSchemalessClient(kubeConfig rest.Config, clientConfig ClientConfig) *SchemalessClient {
```

Comment:
We should make sure we're using the correct NegotiatedSerializer for the underlying REST clients we generate, otherwise we can get bad/undefined behavior. We do the [same thing in NewClientRegistry](https://github.com/grafana/grafana-app-sdk/blob/a6198fc81f66622f0de678d489d4ada785e56540/k8s/client_registry.go#L14-L15).
```suggestion
func NewSchemalessClient(kubeConfig rest.Config, clientConfig ClientConfig) *SchemalessClient {
	kubeConfig.NegotiatedSerializer = &GenericNegotiatedSerializer{}
	kubeConfig.UserAgent = rest.DefaultKubernetesUserAgent()
```